book/urban/sensors/urban-sensors-ukv.ipynb

```diff
@@ -0,0 +1,397 @@
+{
```

Comment:
<p>It's ok to fetch local data but where possible we should show how to fetch from remote URLs. I've tried the lines below but it returns a corrupted file. </p><p> </p><pre class="ql-syntax" spellcheck="false">base_path = "https://metdatasa.blob.core.windows.net/covid19-response-non-commercial/index.html#metoffice_ukv_daily--t1o5m_mean/"
url= base_path + "ukv_daily_t1o5m_mean_20150801.nc"

filename = url.split('/')[-1]
with fsspec.open(url) as f:
&nbsp;&nbsp;with Path(filename).open('wb') as handle:
&nbsp;&nbsp;&nbsp;&nbsp;handle.write(f.read())
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/acocac/environmental-ai-book/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Reply:
Sorry I don't have much experience with doing this kind of thing. Might be worth trying this link - https://metdatasa.blob.core.windows.net/covid19-response-non-commercial/metoffice_ukv_daily/t1o5m_mean/ukv_daily_t1o5m_mean_20150801.nc

Reply:
<p>I believe I have found a compromise by using urllib to retrieve the netcdf file into the sample_data directory, i.e:</p><p> </p><pre class="ql-syntax" spellcheck="false">filepath = 'https://metdatasa.blob.core.windows.net/covid19-response-non-commercial/metoffice_ukv_daily/t1o5m_mean/'
filename = 'ukv_daily_t1o5m_mean_20150801.nc'
urllib.request.urlretrieve(filepath+filename, '../sample_data/' + filename)

</pre><p> </p><p>I think there are other ways of doing this (e.g. curl or wget), but does assume they are installed on the users system. At least using urllib we can specify it as a dependency in the environment.yml</p>

Reply:
<p>excellent! urllib request is a nice one. </p>